sim/battle-actions.ts

```diff
@@ -1217,6 +1214,9 @@ export class BattleActions {
 					}
 				}
 			}
+			if (moveData.selfdestruct === 'ifHit' && damage[i] !== false) {
```

Comment:
I'm checking `damage[i]` before it gets modified based on `didSomething` though, so Healing Wish not returning `false` shouldn't matter.

Reply:
Oh you're talking about that. The "solution" that's being used for that right now is to delete `selfdestruct` if Healing Wish/Lunar Dance fails, which may not be the most elegant but it works (and the action of deleting `selfdestruct` has been done for quite some time). I can't check `damage[i]` after it gets modified by `didSomething` because then Memento will fail on Clear Body when it shouldn't.